dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/ds3treetable/Ds3TreeTablePresenter.java

```diff
@@ -214,41 +220,56 @@ protected Object call() throws Exception {
-                    if (e instanceof FailedRequestException) {
-                        LOG.error("Failed to delete files" + e);
-                        Platform.runLater(() -> deepStorageBrowserPresenter.logText("Failed to delete files : " + ((FailedRequestException) e).getError().getMessage(), LogType.ERROR));
-                        ALERT.setContentText("Failed to delete a files");
-                        ALERT.showAndWait();
-                    } else {
-                        LOG.error("Failed to delete files" + e);
-                        Platform.runLater(() -> deepStorageBrowserPresenter.logText("Failed to delete files", LogType.ERROR));
-                        ALERT.setContentText("Failed to delete a files");
-                        ALERT.showAndWait();
-                    }
+                } catch (final FailedRequestException fre) {
+                    LOG.error("Failed to delete files", fre);
+                    Platform.runLater(() -> deepStorageBrowserPresenter.logText("Failed to delete files : " + fre, LogType.ERROR));
+                    ALERT.setContentText("Failed to delete a files");
+                    ALERT.showAndWait();
+                } catch (final IOException ioe) {
+                    LOG.error("Failed to delete files", ioe);
+                    Platform.runLater(() -> deepStorageBrowserPresenter.logText("Failed to delete files: " + ioe, LogType.ERROR));
+                    ALERT.setContentText("Failed to delete a files");
+                    ALERT.showAndWait();
                 }
                 return null;
             }
         };
-        DeleteFilesPopup.show(task, null, this);
-        values.stream().forEach(file -> refresh(file.getParent()));
-        ds3TreeTable.getSelectionModel().clearSelection();
-        ds3PanelPresenter.getDs3PathIndicator().setText("");
+        DeleteFilesPopup.show(task, null, this, ds3Common);
+       /* values.stream().forEach(file -> refresh(file.getParent()));
+        ds3TreeTable.getSelectionModel().clearSelection();*/
     }
 
     private void createFolderPrompt() {
-        final ImmutableList<TreeItem<Ds3TreeTableValue>> values = ds3TreeTable.getSelectionModel().getSelectedItems()
+        ImmutableList<TreeItem<Ds3TreeTableValue>> values = ds3TreeTable.getSelectionModel().getSelectedItems()
                 .stream().collect(GuavaCollectors.immutableList());
+        final TreeItem<Ds3TreeTableValue> root = ds3TreeTable.getRoot();
         if (values.isEmpty()) {
             deepStorageBrowserPresenter.logText("Select bucket/folder where you want to create an empty folder.", LogType.ERROR);
             ALERT.setContentText("Location is not selected");
             ALERT.showAndWait();
             return;
+        } else if (values.isEmpty()) {
+            final ImmutableList.Builder<TreeItem<Ds3TreeTableValue>> builder = ImmutableList.builder();
+            values = builder.add(root).build().asList();
```

Comment:
If this is just adding a single value this can be simplified to:

```java

values = ImmutableList.of(root);
```

I'm also not sure that the `asList()` call is necessary.